### ISSUE TITLE
InterVM text copy-paste fails when dealing with large texts

### ISSUE THREAD

**Issue by JapanQubes** [QubesOS/qubes-issues#2804]
#### Affected TemplateVMs: All.

---

### Steps to reproduce the behavior:

1. Open two different AppVMs.
2. Open your favorite browser in one of the AppVMs, and open a text editor in the other.
3. Go to https://archive.org/stream/englishjapanesed00satouoft/englishjapanesed00satouoft_djvu.txt then `CTRL+A` and `CTRL+C` followed by `CTRL+SHIFT+C`.
4. Go to your text editor and then press `CTRL+SHIFT+V` then `CTRL+V`.
5. You should see that only one character was inputed, in other words InterVM text copy-paste fails!

### General notes: None.

---

**Comment by andrewdavidwong** [comment]
There has to be *some* limit on the amount of text that can be handled. In my experience, this limit is quite generous. If you're dealing with quantities of text that large, you may want to consider using `qvm-copy-to-vm` instead.

**Comment by andrewdavidwong** [comment]
We could, however, have better feedback/error handling when something like this occurs.

**Comment by JapanQubes** [comment]
> There has to be some limit on the amount of text that can be handled. In my experience, this limit is quite generous.

Right, but I don't think it's generous enough. Even for people dealing with large code projects.

> If you're dealing with quantities of text that large, you may want to consider using `qvm-copy-to-vm` instead.

The only problem with that is that it will be pretty slow to make a single copy-paste across VMs.

**Comment by andrewdavidwong** [comment]
If you're moving such large quantities of text between two VMs, it might be worth asking whether they should really be distinct VMs. There may be a good reason for this (e.g., they need to be based on different TemplateVMs for development reasons), or it may be a good reason to combine them.

**Comment by marmarek** [comment]
For the record: the limit is about 64KB

-- 
Best Regards,
Marek Marczykowski-Górecki
Invisible Things Lab
A: Because it messes up the order in which people normally read text.
Q: Why is top-posting such a bad thing?

**Comment by JapanQubes** [comment]
@andrewdavidwong 

> If you're moving such large quantities of text between two VMs, it might be worth asking whether they should really be distinct VMs. There may be a good reason for this (e.g., they need to be based on different TemplateVMs for development reasons), or it may be a good reason to combine them.

One is a vault-like VM, and the other is a whonix VM. I can edit that vault-VM's settings to enable sys-firewall and then open the  browser in the vault-VM but that would be too cumbersome that's why I choose to just to the copy-paste.

**Comment by cfcs** [comment]
A pop-up when your copy gets truncated would not be a bad idea IMHO.
